src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,829 @@
+            var newStatements = SpecializedCollections.SingletonEnumerable(editor.Generator.ReturnStatement(expression.WithoutTrailingTrivia()));
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var typeSymbol = semanticModel.GetTypeInfo(expression, cancellationToken).ConvertedType ?? semanticModel.Compilation.ObjectType;
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol, name: newMethodIdentifier, returnType: typeSymbol);
+
+            var newMethodDeclaration = editor.Generator.MethodDeclaration(newMethod, newStatements);
+            editor.InsertBefore(containingMethod, newMethodDeclaration);
+        }
+
+        /// <summary>
+        /// Generates a method declaration containing a call to the method that introduced the parameter.
+        /// Example:
+        /// public void M(int x, int y)
+        /// {
+        ///     int f = x * y; // highlight this expression
+        /// }
+        /// 
+        /// ---------------------------------------------------->
+        /// 
+        /// public void M(int x, int y)
+        /// {
+        ///     M(x, y, x * y);
+        /// }
+        /// 
+        /// public void M(int x, int y)
+        /// {
+        ///     int f = x * y;
+        /// }
+        /// </summary>
+        private static void GenerateNewMethodOverload(TExpressionSyntax expression,
+            IMethodSymbol methodSymbol, SyntaxNode containingMethod, SyntaxEditor editor)
+        {
+            var generator = editor.Generator;
+            var arguments = generator.CreateArguments(methodSymbol.Parameters);
+
+            // Remove trailing trivia because it adds spaces to the beginning of the following statement.
+            arguments = arguments.Add(generator.Argument(expression.WithoutTrailingTrivia()));
+            var memberName = methodSymbol.IsGenericMethod
+                ? generator.GenericName(methodSymbol.Name, methodSymbol.TypeArguments)
+                : generator.IdentifierName(methodSymbol.Name);
+
+            var invocation = generator.InvocationExpression(memberName, arguments);
+            var newStatements = SpecializedCollections.SingletonEnumerable(invocation);
+
+            if (!methodSymbol.ReturnsVoid)
+            {
+                newStatements = SpecializedCollections.SingletonEnumerable(generator.ReturnStatement(invocation));
+            }
+
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol);
+            var newMethodDeclaration = generator.MethodDeclaration(newMethod, newStatements);
+            editor.InsertBefore(containingMethod, newMethodDeclaration);
+        }
+
+        /// <summary>
+        /// Gets the matches of the expression and replaces them with the identifier.
+        /// Special case for the original matching expression, if its parent is a LocalDeclarationStatement then it can
+        /// be removed.
```

Comment:
this is a 'what' statement, not really a 'why' statement.  What statements are less helpful.  a 'why' statement helps inform the next person who needs to work in this code.  For example, they will likely ask "why can a local declaration statement be removed?"  "why does it need a rename otherwise, but not here?"

Reply:
answered the "why"